### PR TITLE
mmaplarge.py: added new test case

### PR DESCRIPTION
**Pull request by kalshett**
This test added to check process exit / memory unmap on a large memory system.

Signed-off-by: Kalpana Shetty <kalshett@in.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@kalshett please make pep8 complaint and push that make travis happy :)

File: memory/mmaplarge.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            self.copyutil(file_name)
+
+        build.make(self.teststmpdir)
+
+    def test(self):
+        os.chdir(self.teststmpdir)
+
+        if process.system('./mmaplarge %s' % self.memsize,
+                          shell=True, ignore_status=True):
+            self.fail("Error obtained as expected")
```
Comment (PraveenPenguin) [inline memory/mmaplarge.py]: one help is this message need to changed as `Error obtained as unexpected`

File: memory/mmaplarge.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        self.log.info("process exit/memory unmap tests pass")
```
Comment (PraveenPenguin) [inline memory/mmaplarge.py]: here as well  suited string will be  `exit/memory unmap process  tests pass` as earlier string seems like process got exited

File: memory/mmaplarge.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import os
+import shutil
+from avocado import Test
+from avocado import main
+from avocado.utils import process, build, memory
+from avocado.utils.software_manager import SoftwareManager
+
+
+class process_exit_unmap(Test):
```
Comment (harish-24) [inline memory/mmaplarge.py]: Please use class name with pascal case.

File: memory/mmaplarge.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    memory system
+    """
+
+    def copyutil(self, file_name):
+        shutil.copyfile(self.get_data(file_name),
+                        os.path.join(self.teststmpdir, file_name))
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.memsize = int(memory.meminfo.MemFree.m)
```
Comment (harish-24) [inline memory/mmaplarge.py]: Please use only some percent of free memory as kernel may require some for other tasks.

Comment (harish-24) [inline memory/mmaplarge.py]: As you have already included yaml please use `self.params.get` to get the input from yaml file.

File: memory/mmaplarge.py.data/mmaplarge.c [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	return t->tv_sec*1000000.0 + t->tv_usec;
+}
+
+int main (int argc, char *argv[])
+{
+	unsigned long mapsize = MAPSIZE_DEFAULT;
+	unsigned long mapnum;
+	unsigned long **maps;
+	unsigned long pagespermap = MMAPSIZE/PAGESIZE;
+	char *m;
```
Comment (harish-24) [inline memory/mmaplarge.py.data/mmaplarge.c]: Please consolidate variables based on datatypes.

File: memory/mmaplarge.py.data/mmaplarge.c [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+	/* Maping data */
+	printf("Mapping %liMB worth of pages\n", mapsize/ONEMB);
+	for (i = 0 ; i < mapnum; i++) {
+		maps[i] = mmap(NULL, MMAPSIZE, PROT_READ|PROT_WRITE,
+		  MAP_PRIVATE | MAP_ANON, -1, 0);
+		assert(maps[i]);
+	}
+	/* Reading every page in. This maps to the zero page so no
+		* real memrory used
```
Comment (harish-24) [inline memory/mmaplarge.py.data/mmaplarge.c]: alignment here is improper.

**Comment by PraveenPenguin** [comment]
@kalshett  are you still pursuing this Test case ? if not can we close this

**Comment by PraveenPenguin** [comment]
@kalshett any update on this PR

**Comment by kalshett** [comment]
> @kalshett any update on this PR

Not yet, will close this by Nov 10th, meed to discuss this with Harish.

**Comment by PraveenPenguin** [comment]
@kalshett closing this PR as it is been 6 months ,  if you still persuing this Pull request Please raise new PR having address review comment

**Pull request closed without merging**